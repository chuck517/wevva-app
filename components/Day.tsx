import React from 'react';
import { Text, View, ActivityIndicator } from 'react-native';
import Images from '../assets/index.js';
import LottieView from 'lottie-react-native';
import { styles } from '../styles/styles';

export default function Day({ name, icon, headline, minTemp, maxTemp }) {
  return (
    <View style={styles.forecastList}>
      <View style={styles.forecastListLeft}>
        {icon ? (
          <LottieView
            style={{ height: 50, width: 50, marginRight: 10 }}
            source={Images[icon]}
            autoPlay
            loop
          />
        ) : (
          <ActivityIndicator size="small" color="#fff" />
        )}
        <Text style={styles.forecastText}>
          {name} - {headline}
        </Text>
      </View>
      <View style={styles.forecastListRight}>
        <Text style={styles.forecastText}>
          {minTemp} / {maxTemp}Â°C
        </Text>
      </View>
    </View>
  );
}
