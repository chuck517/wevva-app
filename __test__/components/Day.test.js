import React from 'react';
import renderer from 'react-test-renderer';
import Day from '../../components/Day';


it('renders without crashing', () => {
  const rendered = renderer.create(<Day />).toJSON();
  expect(rendered).toBeTruthy();
});

it('test against snapshot', () => {
  const tree = renderer.create(<Day />).toJSON();
  expect(tree).toMatchSnapshot();
});