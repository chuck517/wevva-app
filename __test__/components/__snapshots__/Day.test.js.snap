// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Day /> renders correctly across screens 1`] = `
<View
  style={
    Object {
      "color": "#fff",
      "flexDirection": "row",
      "height": 55,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 55,
        "width": "75%",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#fff"
      hidesWhenStopped={true}
      size="small"
    />
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 17,
        }
      }
    >
       - 
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 55,
        "justifyContent": "center",
        "width": "25%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 17,
        }
      }
    >
       / 
      °C
    </Text>
  </View>
</View>
`;

exports[`test against snapshot 1`] = `
<View
  style={
    Object {
      "color": "#fff",
      "flexDirection": "row",
      "height": 55,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 55,
        "width": "75%",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#fff"
      hidesWhenStopped={true}
      size="small"
    />
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 17,
        }
      }
    >
       - 
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 55,
        "justifyContent": "center",
        "width": "25%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 17,
        }
      }
    >
       / 
      °C
    </Text>
  </View>
</View>
`;
